core/metamod.cpp

```diff
@@ -417,11 +417,17 @@ FileSystemFactory(const char *iface, int *ret)
 void
 mm_LogMessage(const char *msg, ...)
 {
+	static bool g_logging = false;
+	if (g_logging) {
+		return;
+	}
+	g_logging = true;
 	va_list ap;
 	static char buffer[2048];
 
 	va_start(ap, msg);
 	size_t len = vsnprintf(buffer, sizeof(buffer) - 2, msg, ap);
+	len = std::min<size_t>(len, 2046);
```

Comment:
Where does this magic number come from?

Reply:
Just above, `sizeof(buffer) - 2` though regardless I shouldn't have hard programmed it.

Issue is with the return value of `vsnprintf`, just below we do this.
```
buffer[len++] = '\n';
buffer[len] = '\0';
```
Which fine so as long as `vsnprintf` never returns a number above 2046. However while it's true `vsnprintf` returns the amount of bytes written, it also has another scenario as per c++ documentation : 
> Return Value :
> 4) Number of characters written if successful or negative value if an error occurred. If the resulting string gets truncated due to buf_size limit, function returns the total number of characters (not including the terminating null-byte) which would have been written, if the limit was not imposed.

So if it returns 2047 or above, that's an instant crash when we do
```
buffer[len++] = '\n';
buffer[len] = '\0';
```

Anyways gonna change the value to `sizeof(buffer) - 2`, so there's no magic number.